kernel/src/grant.rs

```diff
@@ -273,6 +282,15 @@ impl<'a, T: Default> Iterator for Iter<'a, T> {
                 return res;
             }
         }
+        // After running through all real apps, pass the debug app (idx 255) to
```

Comment:
(Location doesn't matter. You can only leave review comments on code that has changed.)

I think the `each` implementation for Grant (line 240) needs to be changed as well, since it iterates through app IDs first, which means it will never cover the debug application (or other future kernel apps, presumably)

Reply:
OK take a look at that